qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -60,6 +61,7 @@ def __init__(
         batch_size: int = 900,
         quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
         training_parameters: Optional[Union[ParameterVector, Sequence[Parameter]]] = None,
+        evaluate_duplicates: str | None = "non_diagonal",
```

Comment:
Should this just be of type `str`? If I pass in `None` it looks like it will work, since it does a `str(None)` which ends up as `"None"` and then lowercase makes it the same as passing `"none"` in. I guess having both I find confusing - I mean the None/Optional aspect here and the "none" documented string value. (In the unit tests you always pass in "none" as a string.)

On a side note, would we want to mix/match in the same function both the Union the new | way of doing things. I think it would be better to be consistent at least at this level, though I realize we are moving towards the newer way of doing things where some modules may now be in new style.

Reply:
1. I wanted to show that `None` is a valid value as well. But I fine to keep just `str` as the type hint as well. Should I keep only `str`?
2. I changed the type hints across the class to the new way.

Reply:
OK, let me summarize:

QuantumKernel() -> evaluate_duplicates = off_diagonal
QuantumKernel(evaluate_duplicates=None) -> evaluate_duplicates = none
QuantumKernel(evaluate_duplicates="none") -> evaluate_duplicates = none

This is how it works now and this is exactly what I wanted to achieve. If you think is a wrong way, let me know why.

Reply:
> I wanted to show that None is a valid value as well. But I fine to keep just str as the type hint as well. Should I keep only str?

My preference would be to get rid of None and have it just as `str` type taking "all", "off_diagonal" or "none". None would feel to me it ought to be something different, but it ends up as just a synonym for "none" if you will so to me dropping it seems the thing to do.

Reply:
OK, removed `None`.